### ISSUE TITLE
Bug: Import error when using tsconfig `module: Node16` or newer

### ISSUE THREAD

**Issue by zirkelc** [langchain-ai/langchainjs#4060]
I have a Node.js project with module format `commonjs`, I noticed the following errors when import LangChain packages:

```ts
import { ChatOpenAI } from "@langchain/openai";
import { ChatPromptTemplate } from "@langchain/core/prompts";
import { StringOutputParser } from "@langchain/core/output_parsers";

/* 
The current file is a CommonJS module whose imports will produce 'require' calls; 
however, the referenced file is an ECMAScript module and cannot be imported with 'require'. 
Consider writing a dynamic 'import("@langchain/openai")' call instead.
To convert this file to an ECMAScript module, change its file extension to '.mts', 
or add the field `"type": "module"` to package.json
*/
```

This issue seems to be related to TypeScripts module resolution. If I use `"module": "Node16"` or `"module": "NodeNext"`, the issue appears. If I switch to the very old `"module": "CommonJS"`, the issue disappears. 

I have seen similar issues here with the recommendation to use `require("@langchain/openai")` instead of `import`. 
However, I really don't think that is the real issue because LangChain includes both ESM and CJS packages and it should not be any problem to import CJS in TypeScript via `import`. 

Here is a minimal reproducible example:
https://github.com/zirkelc/langchain-node16-issue

Changing the `"module"` in the tsconfig.json will toggle the import error: 
```jsonc
{
  "compilerOptions": {
    // "module": "CommonJS", // => works
    // "module": "NodeNext", // => doesn't work
    // "module": "Node16", // => doesn't work
    "target": "es2022",
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
  },
  "include": [
    "index.ts",
  ],
  "exclude": [
    "node_modules/**/*",
  ],
}
```

**Comment by jacoblee93** [comment]
Thank you for the report. Will dig in as soon as I can.

**Comment by zirkelc** [comment]
Hi @jacoblee93,
I encountered the same issue for NanoID awhile ago: https://github.com/ai/nanoid/issues/422

Gladly, they were able to fix it by: https://github.com/ai/nanoid/issues/422#issuecomment-1777877914

**TLDR**: the package.json `exports` need to reference `.js` and `.d.ts` files for both ESM and CJ. That means, exporting `.js` and `.d.ts` for ESM, and `.cjs` and `.d.cts` for CJS. See [TS reference](https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-exports)

I was able to verify locally by changing the package.json of `@langchain/openai` and adding a `index.d.cts` type file:

Before:
```json
  "exports": {
    ".": {
      "types": "./index.d.ts",
      "import": "./index.js",
      "require": "./index.cjs"
    },
    "./package.json": "./package.json"
  },
```

After:
```json
  "exports": {
    ".": {
      "require": {
        "types": "./index.d.cts",
        "default": "./index.cjs"
      },
      "import": {
        "types": "./index.d.ts",
        "default": "./index.js"
      }
    },
    "./package.json": "./package.json"
  },
```

The new `index.d.cts` is a copy of the existing `index.d.ts`:

```ts
export * from './dist/index.js'
```

I hope this helps you. Let me know if I can do anything else!

**Comment by jacoblee93** [comment]
That's extremely helpful! Thanks so much.

I think with all of the integration packages we're splitting out we should get some kind of scaffolding lib in place along with this to avoid excessive copy paste...

**Comment by thibautvdu** [comment]
I'm facing the same issue, and unfortunately I have to stick to NodeNext for module resolution. Any work around ?

**Comment by zirkelc** [comment]
@thibautvdu it's a TS-only error, so you can `@ts-ignore` these errors as a workaround.

**Comment by jacoblee93** [comment]
This should be fixed in the latest versions of `langchain`, `@langchain/core`, `@langchain/community`, and `@langchain/openai`! Please reopen if it persists.

**Comment by thibautvdu** [comment]
Hi @jacoblee93 thanks a lot for the update ;

The issue is solved for the aforementioned packages, but is still present with @langchain/misralai

**Comment by jacoblee93** [comment]
Will push a new version of all integration packages today and update you 👍

**Comment by zirkelc** [comment]
Hi @jacoblee93 

this issue is mainly fixed, but I noticed the `@langchain/openai` still has an incomplete exports declaration:

https://github.com/langchain-ai/langchainjs/blob/c80d51835832968b4341445084413511a0c079f8/libs/langchain-openai/package.json#L85-L94

Will you going to change this package too?

**Comment by jacoblee93** [comment]
Oh crap thanks. Will patch today.

**Comment by jacoblee93** [comment]
Hey, this slipped my mind. Fixing now.

**Comment by jacoblee93** [comment]
Wait no @zirkelc I think that is the correct and expected output at this point? We didn't go with exactly what was suggested above.

If it's still causing problems, let me know and we can reopen.
